### ISSUE TITLE
Another tab for related anime

### ISSUE THREAD

**Issue by pitastrudl** [erengy/taiga#119]
How does adding a new tab for related animes sound?

Example:
You finish a series or add one to your Watch list, and the program updates the "Related" tab that shows the movies,specials related to it.

cheers

**Comment by erengy** [comment]
I agree that the feature would be quite useful. The problem is, Taiga doesn't know which anime is related to another. Neither MyAnimeList nor Hummingbird provides the necessary data in their public APIs to third-party applications. Therefore it's not possible to display related anime, unless the application manually creates, manages and distributes its own relation database. It works for web applications such as MALgraph, but it's not exactly a good idea for desktop applications such as Taiga.

**Comment by Soinou** [comment]
Best way to do it would be to create a Taiga-specific server containing these relation databases, but it would mean a lot of work, and possibly users to fill these databases, implying some kind of service making Taiga users able to add relations and everything, but again, this means a lot more work.

But it would be an interesting feature.

**Comment by rr-** [comment]
I'm interested in providing such "static API".

**Comment by Soinou** [comment]
I don't know how MAL api works, but is there no possibility to get informations from there?

For example recommendations, adaptations, prequels and sequels?

Should make everything a lot easier maybe.

**Comment by rr-** [comment]
Nope. MAL API is a joke.

**Comment by Soinou** [comment]
Well, then developing an external API maybe a bit more useful than MAL's one doesn't seem like a bad idea. Even if it means starting from scratch and retrieving informations from another API, then adding extras on it, it could probably help a lot of projects like Taiga.

**Comment by rr-** [comment]
Actually there was such a thing, and it was located at mal-api.com, but it's no more. I'm thinking of reviving it in more concise form (such as no list updates, only static data.)

**Comment by Soinou** [comment]
Yes, but I meant you could be able to get at least some informations from MAL, since the API is able to get a bit of anime informations.

**Comment by pitastrudl** [comment]
I got the idea when I saw the tab upon opening of the program "MAL Updater", im not sure how they did it, but there's a "Related" tab there. I'm not implying anything, just saying.

**Comment by erengy** [comment]
@pitastrudl
I imagine that MAL Updater has a database of its own, distributed along with the application. This is actually very easy to do. I already have a recent database dump myself, which includes relations. However:

- It requires constant effort to keep the database up to date.
- I'm not fond of scraping HTML pages to collect the information.
- The larger the database gets, the more unethical it becomes to distribute it without MAL's consent.

Taiga already distributes a [partial database](https://github.com/erengy/taiga/tree/master/data/db/season) of ~2000 entries for the Seasons page, so the things I said above may sound hypocritical. But I feel I need to draw a line somewhere.

@Soinou
I'm trying to avoid turning Taiga into a service. Otherwise I could do all kinds of things, including launching an anime list service of my own. I'm also trying to avoid making Taiga rely on unofficial APIs. This way, even if it lacks some features, Taiga continues to work when other applications fail (this happened many times in the past, Taiga users didn't notice). That said, I agree that there should be a better API around.

@rr-
There's [another unofficial API](http://bitbucket.org/ratan12/atarashii-api), though I think it requires you to deploy it yourself. This is currently used by a couple of desktop and mobile applications.

Here's what I think would make sense to have in an alternative anime API:

What MAL API must have had:

- A method to retrieve anime information by ID
- More anime metadata (age rating, average user rating, episode length, genres, producers, relations)
- A reliable search method (i.e. one that doesn't fail just because the query is "Fate stay night" rather than "Fate/stay night")
- Advanced searching capabilities (with paginated results, if needed)

What MAL API is probably never going to have:

- Daily database dumps (synopses can be omitted to reduce the file size by half; in which case the database in JSON format would be ~2MB, compressed)
- Airing times of TV series (i.e. an array of MAL IDs coupled with the airing time of their next episodes)
- Episode listing and information (i.e. the title, airing date and synopsis of each episode)
- Mapping of IDs between services (MyAnimeList, Hummingbird, AniList, AniDB, TVDB, etc.)

**Comment by rr-** [comment]
>A reliable search method (i.e. one that doesn't fail just because the query is "Fate stay night" rather than "Fate/stay night")

That sounds like a job for anitomy. Ruby/Python/PHP bindings when? :P (just kidding.)

**Comment by Soinou** [comment]
I understand that you don't want to rely on unofficial APIs, but it could be possible to just use these unofficial APIs for "extras".

I mean, when we talk about the related animes tab, why not get some data from an unofficial API, like the one rr- talked about, and generate a related animes list from this ?

But if the service is not available, then just print a "Not available" message or something like that.

This way, some cool features could be implemented, with the help of an API providing things MAL doesn't have, while letting Taiga work as it worked before.

Well, this is only if the API will exist one day, but for now it's totally out of the question to add these features without any support.

Oh and btw, sorry if I'm not clear, I tend to explain things how I understand it, but not always how others will be able understand it :(

**Comment by erengy** [comment]
@rr-
An alias database could be useful, but I don't think it's relevant in this case. What I meant was searching for user input, and it's just that MAL API's [search method](http://myanimelist.net/modules.php?go=api#animemangasearch) is bad at approximate string matching. The search bar on the website isn't the best either (the results are sorted alphabetically rather than by similarity), but at least it doesn't trip over missing punctuation.

@Soinou
Yes, keeping them optional would be the way to go, so that they don't interfere with the main functionality of the application.

**Comment by pitastrudl** [comment]
@erengy thank you for the explenation.i understand

**Comment by erengy** [comment]
We had some nice discussion here on this issue, but I'm closing it since there's currently no viable solution to implement the feature.

**Comment by darkred3** [comment]
@erengy Since mal is getting a new API, do you know if this feature will be supported?

**Comment by cu1tur4l** [comment]
@erengy 
Sorry for gravedigging, however a lot of time has passed since 2015 so I would love to know if now it would be possible to add such a feature.
I hope you are doing well, since you've been a little inactive on git.
Thank you for creating and maintaining Taiga <3

**Comment by erengy** [comment]
It's been almost ten years, huh. Time flies. 

I think all APIs provide relations data now (e.g. `relations` field in AniList API, `related_anime` field in MAL API). Not sure about how it'd work in practice, but might be possible to show a list of sequels, etc. on the media dialog in v2.

Haven't been able to direct much energy to Taiga in the past few months due to some other priorities in life. I'm doing okay, though. Thanks, @cu1tur4l!
